avocado/utils/kernel.py

```diff
@@ -95,11 +99,26 @@ def build(self):
         Build kernel from source.
         """
         log.info("Starting build the kernel")
+        build_output_format = ""
+        if self.do_install is not None:
+            if self.distro.name == "Ubuntu":
+                build_output_format = "deb-pkg"
         if self.config_path is None:
-            build.make(self.linux_dir, extra_args='O=%s defconfig' % self.build_dir)
+            build.make(self.linux_dir, extra_args='-C %s defconfig' % self.linux_dir)
         else:
-            build.make(self.linux_dir, extra_args='O=%s olddefconfig' % self.build_dir)
-        build.make(self.linux_dir, extra_args='O=%s' % self.build_dir)
+            build.make(self.linux_dir, extra_args='-C %s olddefconfig' % self.linux_dir)
+        build.make(self.linux_dir, extra_args='-C %s %s' % (self.linux_dir, build_output_format))
+
+    def install(self):
+        """
+        Install built kernel.
+        """
+        if self.do_install is not None:
+            log.info("Starting kernel install")
+            if self.distro.name == "Ubuntu":
+                process.run('sh -c "exec sudo dpkg -i %s/*.deb"' % self.work_dir)
+        else:
+            log.info("Skipping kernel install")
```

Comment:
It'd be nice to also include generic kernel installation code.  There's no need to guarantee that it would work on every single system out there, but it feels wrong to add a method that will only work on a given system/package type.

Reply:
We (here at AMD) planned to extend the functionality to other distribution, as we included them in our test-infrastructure; we just happened to have started working with Ubuntu.

Reply:
OK, that's fine.